vl-convert-rs/src/converter.rs

```diff
@@ -218,24 +224,41 @@ import('{url}').then((sg) => {{
 
             // Create and initialize svg function string
             let function_str = r#"
-function vegaToView(vgSpec) {
+function vegaToView(vgSpec, allowedBaseUrls, errors) {
     let runtime = vega.parse(vgSpec);
-    const baseURL = 'https://vega.github.io/vega-datasets/';
+    let baseURL = 'https://vega.github.io/vega-datasets/';
     const loader = vega.loader({ mode: 'http', baseURL });
+    const originalHttp = loader.http.bind(loader);
+
+    if (allowedBaseUrls != null) {
+        loader.http = async (uri, options) => {
+            const parsedUri = new URL(uri);
+            if (
+                allowedBaseUrls.every(
+                    (allowedUrl) => !parsedUri.href.startsWith(allowedUrl),
+                )
+            ) {
+                errors.push(`External data url not allowed: ${uri}`);
```

Comment:
Vega swallows the exception thrown below, so we explicitly keep an array of errors to check after the conversion has finished